File: app/src/main/java/com/jay/currencyconverter/di/BankActivityModule.java
package com.jay.currencyconverter.di;

import com.jay.currencyconverter.ui.adapter.BankExchangeRateAdapter;
import com.jay.currencyconverter.ui.bankActivity.BankPresenter;
import com.jay.currencyconverter.ui.bankActivity.IBankView;

import dagger.Module;
import dagger.Provides;

@Module
public class BankActivityModule {

    @Provides
    public BankExchangeRateAdapter provideBankExchangeRateAdapter() {
        return new BankExchangeRateAdapter();
    }

    @Provides
    public static BankPresenter provideBankPresenter(IBankView view) {
        return new BankPresenter(view);
    }
}

File: app/src/main/java/com/jay/currencyconverter/di/BankActivityModule.kt
package com.jay.currencyconverter.di

import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModelProviders
import com.jay.currencyconverter.ui.adapter.OrganizationExchangeRateAdapter
import com.jay.currencyconverter.ui.organizationActivity.OrganizationActivityViewModel
import dagger.Module
import dagger.Provides

@Module
class BankActivityModule {

    @Provides
    fun provideBankExchangeRateAdapter(): OrganizationExchangeRateAdapter {
        return OrganizationExchangeRateAdapter()
    }

    companion object {
        @Provides
        fun provideBankActivityVM(activity: FragmentActivity): OrganizationActivityViewModel {
            return ViewModelProviders.of(activity).get(OrganizationActivityViewModel::class.java)
        }
    }
}